core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -481,8 +481,14 @@ object SparkSubmit {
     }
 
     // Let YARN know it's a pyspark app, so it distributes needed libraries.
-    if (clusterManager == YARN && args.isPython) {
-      sysProps.put("spark.yarn.isPython", "true")
+    if (clusterManager == YARN) {
+      if (args.isPython) {
+        sysProps.put("spark.yarn.isPython", "true")
+      }
+      if (args.principal != null) {
+        require(args.keytab != null, "Keytab must be specified when the keytab is specified")
```

Comment:
so theoretically a user could specify spark.yarn.keytab but then use --principal on the command line or vise versa.  Currently this doesn't allow that. For completeness it might be nice to handle any combination of those.  Thoughts?